From b4a1bafc949d870fba384cfa4374fe075193e1ff Mon Sep 17 00:00:00 2001
From: Tianshu Qiu <tian.shu.qiu@intel.com>
Date: Tue, 14 Apr 2015 18:33:41 +0800
Subject: [PATCH 354/360] atomisp2: clean parameters after css is stopped.

This patch cleans the css parameters when it is stopped.
This can aviod the previous parameters being used for the
next streaming.

Change-Id: Icb790faee18e8a7610c3093e43cafd0558449bd3
Fixes-Issue: #JIMINAN-29878
Signed-off-by: Tianshu Qiu <tian.shu.qiu@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/51279
Approver: Bin B Han <bin.b.han@intel.com>
Reviewed-by: Bin B Han <bin.b.han@intel.com>
Integrator: Wei Tao <wei.tao@intel.com>
Validation-Android: Wei Tao <wei.tao@intel.com>
Build: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
Maintainer: Wei Tao <wei.tao@intel.com>
---
 .../drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c     | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
index ab684bb..c9beff8 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
@@ -2406,6 +2406,7 @@ int atomisp_css_stop(struct atomisp_sub_device *asd,
 	atomisp_flush_params_queue(&asd->video_out_vf);
 	atomisp_flush_params_queue(&asd->video_out_preview);
 	atomisp_flush_params_queue(&asd->video_out_video_capture);
+	memset(&asd->params.css_param, 0, sizeof(asd->params.css_param));
 	return 0;
 }
 
-- 
1.9.1

